app/lib/searching.rb

```diff
@@ -0,0 +1,265 @@
+class Searching
+  require 'gds_api/rummager'
+  attr_reader :params
+  def initialize(params)
+    @params = params
+  end
+
+  def count
+    return 10 if params["count"].blank?
+    return 1000 if params["count"].to_i > 1000
+    params["count"]
+  end
+
+  def call
+    terms = @params["search_term"].split.join("+")
+    rummager = GdsApi::Rummager.new(Plek.new.find('rummager'))
+    findings_new_left = rummager.search(q:      "#{terms}",
+                                        fields: [ "is_historic", "title", "link", "popularity", "public_timestamp", "description",
+                                                  "format", "content_id", "document_collections", "mainstream_browse_pages",
+                                                  "organisations", "taxons", "people", "policies", "specialist_sectors"],
+                                        count:  "#{count}", ab_tests: "#{params["which_test"]}:A", c: "#{Time.now.getutc}")
+    findings_new_right = rummager.search(q:      "#{terms}",
+                                        fields: [ "is_historic", "title", "link", "popularity", "public_timestamp", "description",
+                                                  "format", "content_id", "document_collections", "mainstream_browse_pages",
+                                                  "organisations", "taxons", "people", "policies", "specialist_sectors"],
+                                        count:  "#{count}", ab_tests: "#{params["which_test"]}:B", c: "#{Time.now.getutc}")
+    Results.new(findings_new_left, findings_new_right)
+  end
+
+  class Results
+    attr_reader :results, :left_total, :left_missing, :right_total, :right_missing
+    def initialize(left, right)
+      result_count = right['results'].count > left['results'].count ? right['results'].count : left['results'].count
+      @results = (0..result_count-1).map { |i| Result.new(left, right, i) }
+      @left_total = left['total']
+      @right_total = right['total']
+      @left_missing = @left_total - left['results'].count
+      @right_missing = @right_total - right['results'].count
+    end
+  end
+
+  class Result
+    attr_reader :left, :right, :position, :enhanced_results_hash
+    def initialize(left, right, position)
+      @left = left['results'].each_with_index.map {|r, i| [r, i]}
+      @right = right['results'].each_with_index.map {|r, i| [r, i]}
+      @position = position
+      @date_hash = {
+              "01" => "January",
+              "02" => "February",
+              "03" => "March",
+              "04" => "April",
+              "05" => "May",
+              "06" => "June",
+              "07" => "July",
+              "08" => "August",
+              "09" => "September",
+              "10" => "October",
+              "11" => "November",
+              "12" => "December" }
+      @enhanced_results_hash = {}
+    end
+
+    def content_id(side)
+      begin
+        side[position][0]['content_id']
+      rescue
```

Comment:
blank rescues are dangerous as they hide unexpected errors. 

Also we can easily achieve the same with a check condition:

```
def content_id(side)
  if side[position]
    side[position][0]['content_id']
  else
    nil
  end
end
```